### ISSUE TITLE
Change built-in constructor inheritance chains to match ES6

### ISSUE THREAD

**Issue by svaarala** [svaarala/duktape#921]
For example, `TypeError` _constructor_ inherits (internal prototype) from `Error` constructor. In ES5.1 `TypeError` inherits from `Function.prototype`. See also #918.

**Comment by fatcerberus** [comment]
Changing the prototype doesn't affect callability?  In Duktape specifically I mean, I know other engines (V8, at least) handle this fine. :)

**Comment by svaarala** [comment]
Duktape is no different in that respect:

```
((o) Duktape [linenoise] 1.99.99 (v1.5.0-456-g60f7827-dirty)
duk> function foo() { print('called'); }
= undefined
duk> Object.setPrototypeOf(foo, null); foo();
called
= undefined
duk> foo.call
= undefined
```

**Comment by fatcerberus** [comment]
Relevant section of the ES6 specification:
http://www.ecma-international.org/ecma-262/6.0/#sec-properties-of-the-nativeerror-constructors

Internal prototype of TypeError etc. is `%Error%`, i.e. the initial value of `Error`.

**Comment by fatcerberus** [comment]
Surprisingly, V8 seems to be the only engine that honors this.  Firefox doesn't despite implementing most or all of ES6, nor does (mobile) Safari.

**Comment by svaarala** [comment]
This was fixed by #1182 for Errors. I'll check if other built-ins have similar issues.

**Comment by svaarala** [comment]
The following are from ES2016:
- NativeError subclasses, done in #1182
- TypedArray constructors inherit from %TypedArray% intrinsic object (which further inherits from Function.prototype), this needs fixing -> #1191
- TypedArray prototype objects (not DataView) inherit from %TypedArrayPrototype% intrinsic object (which further inherits from Object.prototype), this has already been the case because I skimped a bit on Khronos typed array compliance to reduce footprint ;)
